spec/unit/puppet/provider/mongodb_database/mongodb_spec.rb

```diff
@@ -35,9 +35,9 @@
   before do
     tmp = Tempfile.new('test')
     mongodconffile = tmp.path
-    allow(provider.class).to receive(:mongod_conf_file).and_return(mongodconffile)
-    allow(provider.class).to receive(:mongo_eval).with('rs.slaveOk();printjson(db.getMongo().getDBs())').and_return(raw_dbs)
-    allow(provider.class).to receive(:db_ismaster).and_return(true)
+    allow(provider.class).to have_received(:mongod_conf_file).and_return(mongodconffile)
```

Comment:
My understanding of `have_received` was that it's used to assert. So you allow to receive but then expect to have received. https://relishapp.com/rspec/rspec-mocks/docs/basics/spies#spy-on-a-method-on-a-partial-double